pkg/kube/kube_handler_test.go

```diff
@@ -909,6 +910,45 @@ func TestGetNamespaces(t *testing.T) {
 			expectedNamespaces: []string{"bar"},
 			allowed:            true,
 		},
+		{
+			name: "it lists namespaces if the user client sends the namespaces",
+			existingNamespaces: []existingNs{
+				{"foo", corev1.NamespaceActive},
+			},
+			trustedNamespaces: []corev1.Namespace{{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+				Status:     corev1.NamespaceStatus{Phase: corev1.NamespaceActive},
+			}},
+			expectedNamespaces: []string{"foo"},
+			allowed:            true,
+		},
```

Comment:
Similar to the other tests, I'm not sure these tests are testing what you think they're testing. For starters, in your use case, the user client should not be even contacted, as the namespaces are passed in? I don't think you should even be setting `existingNamespaces` for your tests, given that we expect the API server not to be called for the list (especially if you remove the use of `filterAllowedNamespaces` in your code which may cause an indirect query).

I think I'd just have three tests here:

* it returns namespaces from header when specified
* it filters namespaces from header to return only active namespaces
* it returns namespaces from header, despite user and service account forbidden

and for all three, `existingNamespaces` need not be set (as it should be empty).

Reply:
I review the code a few times and if I'm not wrong, the test does the job. I reduced the test to only 2 cases:
- it returns namespaces from header when specified,
- it returns existing namespaces when `whitelistedNamespaces` is empty i.e. header does not match, it's empty, or one of the kubeops arg is empty.